### PR TITLE
Add hadoop 3.2.4 support

### PR DESCRIPTION
**Pull request by tigrulya-exe**
1. Added `smart-hadoop-3.2` and `smart-hadoop-client-3.2` modules;
2. Moved common logic for hdfs 3.x.x into separate module;
3. Rewrote the logic of getting instances of `MiniClusterFactory` in tests. Implementations are now initialized by SPI;
4. Enabled and actualized zeppelin tests.

### CODE REVIEW

File: smart-common/pom.xml [hunk 1]
```diff
@@ -62,6 +62,11 @@
             <groupId>org.apache.logging.log4j</groupId>
             <artifactId>log4j-slf4j-impl</artifactId>
         </dependency>
+        <dependency>
+            <groupId>commons-lang</groupId>
+            <artifactId>commons-lang</artifactId>
+            <version>2.6</version>
```
Comment (xardazzzzzz) [inline smart-common/pom.xml]: add version property

File: smart-dist/pom.xml [hunk 2]
```diff
@@ -183,6 +183,21 @@
                 </dependency>
             </dependencies>
         </profile>
+        <profile>
+            <id>hadoop-3.2</id>
+            <dependencies>
+                <dependency>
+                    <groupId>org.smartdata</groupId>
+                    <artifactId>smart-hadoop-3.2</artifactId>
+                    <version>1.6.0-SNAPSHOT</version>
```
Comment (xardazzzzzz) [inline smart-dist/pom.xml:192]: same here

**Review by xardazzzzzz — APPROVED** [review]

File: smart-common/pom.xml (hunk 1 above)
Comment (tigrulya-exe) [reply smart-common/pom.xml]: Fixed

File: smart-dist/pom.xml (hunk 2 above)
Comment (tigrulya-exe) [reply smart-dist/pom.xml:192]: It's subproject, so I inlined version as it was done for other submodules in project

**Pull request merged**